src/app/auth/services/courses-store.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { CourseModule } from '../../shared/components/course/course.module';
+import { CoursesService } from './courses.service';
+import { tap } from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class CoursesStoreService {
+    private _courses$ = new BehaviorSubject<CourseModule[]>([]);
+    private coursesLoaded = false;
+
+    constructor(private coursesService: CoursesService) {}
+
+    get courses$(): Observable<CourseModule[]> {
```

Comment:
every time accessing this getter there will be another subscription/request, is it intentional?

Reply:
I'd expose `this._courses$` in a way it can be 
```suggestion
  readonly courses$ = of('gettingCources').pipe(
     tap(() => {
       this.coursesLoaded = false
     }),
     switchMap(() => this.coursesService.getAll()),
     finalize(() => this.coursesLoaded = true)
   )
```

Or if you need external trigger to recreate request

```suggestion
    ```suggestion
    readonly coursesFetchTrigger$ = new BehaviourSubject()
  readonly courses$ = this.coursesFetchTrigger$.asObservable().pipe(
     tap(() => {
       this.coursesLoaded = false
     }),
     switchMap(() => this.coursesService.getAll()),
     finalize(() => this.coursesLoaded = true)
   )
```
```